src/mapping/Partition.hpp

```diff
@@ -0,0 +1,315 @@
+#pragma once
+
+#include <Eigen/Core>
+
+#include <boost/container/flat_set.hpp>
+
+#include "mapping/RadialBasisFctSolver.hpp"
+#include "mapping/config/MappingConfiguration.hpp"
+#include "mapping/impl/BasisFunctions.hpp"
+#include "mesh/Filter.hpp"
+#include "precice/types.hpp"
+#include "utils/Event.hpp"
+
+namespace precice {
+
+namespace mapping {
+
+/**
+ * @brief Mapping using partition of unity decomposition strategies
+ */
+template <typename RADIAL_BASIS_FUNCTION_T>
+class Partition {
+public:
+  /**
+   * @brief Constructor.
+   *
+   * @param[in] dimension Dimensionality of the meshes
+   * @param[in] center Center of the particular partition
+   * @param[in] radius radius of the partition
+   * @param[in] parameter Shape parameter or support radius
+   * @param[in] targetSize target size (cardinality) of the partition
+   * @param[in] inputMesh the input mesh
+   * @param[in] outputMesh the output mesh
+   */
+  Partition(int               dimension,
+            mesh::Vertex      center,
+            double            radius,
+            double            parameter,
+            std::vector<bool> deadAxis,
+            Polynomial        polynomial,
+            unsigned int      targetSize,
+            mesh::PtrMesh     inputMesh,
+            mesh::PtrMesh     outputMesh);
+
+  /// Removes a computed mapping.
+  void clear();
+
+  /// Execute a consistent mapping
+  void mapConsistent(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// Execute a consistent mapping
+  void mapConservative(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// @brief set the normalized weight in the normalizedWeight data structure
+  void setNormalizedWeight(double normalizedWeight, VertexID vertexID);
+
+  /// Compute the weight for a given vertex
+  double computeWeight(const mesh::Vertex &v) const;
+
+  /// Evaluate the interpolant for a given vertex
+  double getInterpolatedValue(const VertexID id, unsigned int direction) const;
+
+  /// Indicate whether a vertex lies within the partition or not
+  bool isVertexInside(const mesh::Vertex &v) const;
+
+  /// Indicate that the mapped data is now considered as outdated by setting _hasMappedData to false
+  void setMappingFinished();
+
+  /// Number of input vertices this partition operates on
+  unsigned int getNumberOfInputVertices() const;
+
+  /// The center coordinate of this partition
+  std::array<double, 3> getCenterCoords() const;
+
+  /// Indicates whether there are valid output points for this partition
+  bool isEmpty() const;
+
+private:
+  precice::logging::Logger _log{"mapping::Partition"};
+
+  bool _hasComputedMapping = false;
+  bool _hasMappedData      = false;
+  bool _emptyPartition     = false;
+
+  /// Partition center
+  mesh::Vertex _center;
+
+  /// Partition radius
+  const double _radius;
+
+  /// Array with one entry per data component. A plain array would require knowledge about the number of components
+  std::array<Eigen::VectorXd, 3> resultVector;
```

Comment:
I am not sure I understand what "data component" means? "3" for vector-valued data? Maybe we could also find a better name for "resultVector".

`_resultVector`?

Reply:
..was a an obsolete leftover..